lib/db/leveldb.go

```diff
@@ -408,6 +427,12 @@ func ldbUpdateGlobal(db dbReader, batch dbWriter, folder, device []byte, file pr
 			panic(err)
 		}
 
+		if len(fl.versions) > 0 {
+			// Keep the current neweset file around so we can subtract it from
+			// the globalSize if we replace it.
+			oldFile, hasOldFile = ldbGet(db, folder, fl.versions[0].device, name)
```

Comment:
Thing is that if the previous newest version was ours, and the new newest version is also ours, the old version is gone by the time we need it. But perhaps I could optimize this, let me have a look.